chore: Add issue and PR templates (#1327)
Amsterdam/design-system: .github/pull_request_template.md

# Describe the pull request

> Thank you for contributing to the project! Please take some time to fill out this template, to help us quickly and smoothly handle your PR.

Reviewer: ```suggestion
Thank you for contributing to the project!
Please use this template to help us handle your PR smoothly.
```

## What

> (What changes are being introduced in this PR?)

Reviewer: ```suggestion
(What changes does this PR introduce?)
```

## Why

(Why are these changes necessary?)

## How

(How were these changes implemented? Provide a brief overview of the approach taken and any key considerations.)

## Checklist

Before submitting your pull request, please ensure you have done the following. Check each checkmark if you have done so or if it wasn't necessary:

- [ ] Add or update unit tests
- [ ] Add or update documentation
- [ ] Add or update stories
- [ ] Add or update exports in index.\* files
- [ ] Start the PR title with a Conventional Commit prefix, [as explained here](https://github.com/Amsterdam/design-system/blob/main/documentation/publishing.md?plain=1#L11).

## Additional notes

> - Please link to any related issues or discussions.

Reviewer: ```suggestion
- Provide links to any related issues or discussions.
```

> - Please link to the relevant story in the feature branch deploy.

Reviewer: ```suggestion
- Add a link to the specific story in the feature branch deploy.
```
